linera-core/src/unit_tests/test_utils.rs

```diff
@@ -724,17 +724,18 @@ impl GenesisStorageBuilder {
     }
 }
 
-pub type ChainClient<S> = crate::client::ChainClient<crate::environment::Impl<S, NodeProvider<S>>>;
+pub type ChainClient<S> =
+    crate::client::ChainClient<crate::environment::Impl<S, NodeProvider<S>, InMemorySigner>>;
```

Comment:
Looks like `ChainClient` could use a `with_signer(self)` (maybe not now, but as some point) - that replaces the `Signer` instance while keeping everything else.

Reply:
Feel free to add it as part of the JavaScript signer :) I'm not sure how much sense it makes to change signer after the client has started but it is tempting to be able to set the signer on a per-chain basis.